packages/web3-common/src/web3_base_wallet.ts

```diff
@@ -4,15 +4,32 @@ export interface Web3BaseWalletAccount {
 	[key: string]: unknown;
 	readonly address: string;
 	readonly privateKey: string;
-	readonly signTransaction: (tx: Record<string, unknown>) => string;
-	readonly sign: (data: Record<string, unknown> | string) => string;
-	readonly encrypt: (password: string, options?: Record<string, unknown>) => Web3EncryptedWallet;
+	readonly signTransaction: (tx: Record<string, unknown>) => {
+		readonly messageHash: string;
+		readonly r: string;
+		readonly s: string;
+		readonly v: string;
```

Comment:
Good to create type and reuse that in sign function return type as well.

Reply:
Could use [HexString](https://github.com/ChainSafe/web3.js/blob/bfae214735803e38ba6bddabcf43f241d715c5d2/packages/web3-utils/src/types.ts#L1) from `web3-utils` as it's already being imported in this package